ddtrace/tracer/tracer.go

```diff
@@ -306,16 +316,50 @@ func (t *tracer) worker(tick <-chan time.Time) {
 	}
 }
 
-func (t *tracer) pushTrace(trace []*span) {
+// traceInfo contains span list and sampling decision of the trace
+// needed to run single span sampling.
+type traceInfo struct {
+	spans            []*span
+	samplingDecision samplingDecision
+}
+
+// processTraceInfo runs single span sampling on the spans pushed to the worker thread.
```

Comment:
This explanation is very technical and assumes that the reader has a lot of context. If possible (it's not always possible) we should try to document things in a more independent way, for example:
```go
// sampleFinishedTrace applies single-span sampling to the provided trace, which is considered to be finished.
```